presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -592,6 +593,52 @@ public Node visitResetSession(SqlBaseParser.ResetSessionContext context)
         return new ResetSession(getLocation(context), getQualifiedName(context.qualifiedName()));
     }
 
+    @Override
+    public Node visitGrant(SqlBaseParser.GrantContext context)
+    {
+        String grantee;
+        if (context.PUBLIC() != null) {
+            grantee = "PUBLIC";
+        }
+        else {
+            grantee = context.grantee.getText();
+        }
+
+        List<String> privileges;
+        if (context.ALL() != null && context.PRIVILEGES() != null) {
+            //List all privileges  declared in 'Privilege' enum in SPI
+            privileges = ImmutableList.of("SELECT", "DELETE", "INSERT");
```

Comment:
I talked to @martint about this.  He says the AST builder should create a tree that is a fateful representation of the original SQL, and should "roundtrip" (we consider places we don't do this today to be a problem).  Therefore, he would like to see an "all privilages" flag on the AST node, and tests that verify it rounds trips correctly.